src/main/java/io/stargate/sgv2/jsonapi/api/model/command/table/definition/datatype/VectorColumnDesc.java

```diff
@@ -28,8 +29,8 @@ public VectorizeConfig getVectorizeConfig() {
     return vectorizeConfig;
   }
 
-  public int getDimensions() {
-    return dimensions;
+  public Optional<Integer> getDimensions() {
```

Comment:
as per above, since caller immediate unwraps `Optional` could just return `Integer` as-is.